### PR TITLE
feat: Debian12 readyness

### PR DESCRIPTION
**Pull request by madpat1998**
Fix various things to make the Playbook run on debian 12.

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -153,14 +153,31 @@ icinga2_master_graphite_host: 127.0.0.1
 # The port graphite is listening on
 icinga2_master_graphite_port: 2003
 
-## Icinga2 ido database settings
+## Icinga2db settings (DO NOT USE ICINGADB AND IDO AT THE SAME TIME)
```
Comment (jon4hz) [inline defaults/main.yml:156]: Maybe add an `assert` somewhere to make sure this is enforced.

File: templates/etc/icinga2/features-available/ido-mysql.conf.j2 [hunk 2]
```diff
@@ -14,6 +14,6 @@ object IdoMysqlConnection "ido-mysql" {
   host = "{{ icinga2_master_icinga2_database_host }}"
   database = "{{ icinga2_master_icinga2_database_name }}"
   {% if icinga2_master_icinga2_database_ssl -%}
-  enable_ssl = true
+  enable_ssl = "{{ icinga2_master_icinga2_database_ssl }}"
```
Comment (jon4hz) [inline templates/etc/icinga2/features-available/ido-mysql.conf.j2:17]: This changes the default behavior, is this intended?

File: templates/etc/icinga2/constants.conf.j2 [hunk 3]
```diff
@@ -9,17 +9,17 @@
  */
 
 /* The directory which contains the plugins from the Monitoring Plugins project. */
-const PluginDir = "/usr/lib64/nagios/plugins"
+const PluginDir = "{{ icinga2_master_nagios_plugin_dir }}"
```
Comment (jon4hz) [inline templates/etc/icinga2/constants.conf.j2:12]: `icinga2_master_nagios_plugin_dir` defaults to `/usr/lib/nagios/plugins` and not `/usr/lib64/nagios/plugins`. Intended?

**Comment by jon4hz** [comment]
Maybe we should add antsibull changelogs to point out any breaking changes. What do you think?

File: templates/etc/icinga2/constants.conf.j2 (hunk 3 above)
Comment (gianklug) [reply templates/etc/icinga2/constants.conf.j2:12]: Yes, as the role now targets Debian 12 instead of RH, where this is the default.

Comment (jon4hz) [reply templates/etc/icinga2/constants.conf.j2:12]: In that case, I think we should make the default based on the `ansible_os_family`. 
```suggestion
const PluginDir = "{{ icinga2_master_nagios_plugin_dir | default(_icinga2_master_nagios_plugin_dir) }}"
```
Remove `icinga2_master_nagios_plugin_dir` from the defaults and add `_icinga2_master_nagios_plugin_dir` to the os-specific vars. This way we can make it configurable, but the default works for multiple os families.

Comment (gianklug) [reply templates/etc/icinga2/constants.conf.j2:12]: We decided on deprecating RHEL support; as there are no OS specific vars yet like in other roles, I'm for just keeping it like this.

For roles that support multiple distros, we usually do something like this and include them based on the OS:
https://github.com/adfinis/ansible-role-icinga2_web/tree/master/vars

File: defaults/main.yml (hunk 1 above)
Comment (gianklug) [reply defaults/main.yml:156]: done

**Comment by gianklug** [comment]
> Maybe we should add antsibull changelogs to point out any breaking changes. What do you think?

Not sure, as we're currently also not versioning the role...

**Comment by jon4hz** [comment]
> > Maybe we should add antsibull changelogs to point out any breaking changes. What do you think?
> 
> Not sure, as we're currently also not versioning the role...

Oh well, wasn't aware of that.

**Pull request merged**